nomad/eval_broker.go

```diff
@@ -369,6 +385,17 @@ SCAN:
 		if timeoutTimer != nil {
 			timeoutTimer.Stop()
 		}
+		b.l.Lock()
+		if t, ok := b.enqueuedTime[eval.ID]; ok {
+			b.dequeuedTime[eval.ID] = time.Now()
```

Comment:
We should limit the map size here as well to make sure the map doesn't grow too large.

Reply:
corrected in [268db7f](https://github.com/hashicorp/nomad/pull/20329/commits/268db7fde683a9ab3247c618855b85ae353aa365)